### PR TITLE
Fix possible completed job state issue

### PR DESCRIPTION
**Pull request by LyzeOfKiel**
Fix possible concurrency issue (reported by @EvgenKor).

Because `job.getState` method is asynchronous following scenario is possible:
1. We get the job from queue by id
2. We asynchronously get job's state
3. The state is `completed`
4. Job object that we got in the first step does not contain correct `returnvalue` because it was not in `completed` state at the moment of fetching.

Solution:
If we got `completed` job's state, we need to check job's `returnvalue` and if it is `null` re-fetch the job.

### CODE REVIEW

File: zp-relayer/endpoints.ts [hunk 1]
```diff
@@ -146,8 +146,13 @@ async function getJob(req: Request, res: Response) {
     const poolJobState = await job.getState()
     if (poolJobState === 'completed') {
       // Transaction was included in optimistic state, waiting to be mined
+      if (job.returnvalue === null) {
+        job = await poolTxQueue.getJob(jobId)
+        // Sanity check
+        if (!job) throw new Error('Pool job not found')
+    }
```
Comment (EvgenKor) [inline zp-relayer/endpoints.ts:153]: You fetch job again, but `returnvalue` property of this object is still unchecked. Imagine you have another issue reason (not concurrent nature) which leads `job.returnvalue` equals null (even after second job request). I think here should be some kind of limited-iteration while cycle, or secondary checking `job.returnvalue` with proper internal error returning

**Review by EvgenKor — CHANGES_REQUESTED** [review]

File: zp-relayer/endpoints.ts (hunk 1 above)
Comment (LyzeOfKiel) [reply zp-relayer/endpoints.ts:153]: If we got `completed` job's state there should be no reasons why `returnvalue` could be `null` (possible only if the job returned `null` itself, but there is no such cases in current implementation). I think such check would be redundant.

Comment (LyzeOfKiel) [reply zp-relayer/endpoints.ts]: Added `returnvalue` check

Comment (EvgenKor) [reply zp-relayer/endpoints.ts]: Seems that problem should be covered

**Review by EvgenKor — APPROVED** [review]

**Review by k1rill-fedoseev — APPROVED** [review]

**Pull request merged**